q2cli/cli.py

```diff
@@ -27,6 +27,10 @@ class QiimeCLI(click.MultiCommand):
 
     def list_commands(self, ctx):
         plugins = list(sorted(self._plugin_manager.plugins.keys()))
+        for idx, key in enumerate(plugins):
+            plugin = self._plugin_manager.plugins[key]
+            if not plugin.methods or not plugin.visualizers:
```

Comment:
This will remove plugins from the list if they don't have methods or if they don't have visualizers. We want to remove the plugin only if it has exactly zero methods *and* visualizers.